src/schemas/f142/f142.cpp

```diff
@@ -362,12 +363,25 @@ class Converter : public FlatBufferCreator {
     // this is the field type ID string: up.pvstr->getStructure()->getID()
     auto PVName = Builder->CreateString(PVUpdate.channel);
     auto Value = makeValue(*Builder, PVStructure, true, Stats);
-
     LogDataBuilder LogDataBuilder(*Builder);
     LogDataBuilder.add_source_name(PVName);
     LogDataBuilder.add_value_type(Value.Type);
     LogDataBuilder.add_value(Value.Offset);
 
+    if (auto PVDisplay =
```

Comment:
Could this code block be extracted to a separate method?

Reply:
It's now separated into two methods - one for recording timestamp and one for checking units.